File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d = {}
        for letter in s:
            if letter not in d:
                d[letter]=1
            else:
                d[letter]+=1
        for letter in t:
            if letter not in d:
                return False
            else:
                d[letter]-=1
        for value in d.values():
            if value != 0:
                return False
        return True    
            
                
        
        

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
      public static boolean isAnagram(String s, String t) {
           if (s.length() != t.length()) {
            return false;
        }

        // Convert both strings to character arrays
        char[] sArray = s.toCharArray();
        char[] tArray = t.toCharArray();

        // Sort both character arrays
        Arrays.sort(sArray);
        Arrays.sort(tArray);

        // Compare the sorted arrays
        return Arrays.equals(sArray, tArray);
    }
}
        